bpaotu/bpaotu/query.py

```diff
@@ -173,37 +187,64 @@ def _q_all_cached(self, topic, q, mutate_result=None):
             cache.set(key, result)
         return result
 
+
     def matching_sample_ids_and_environment(self):
         q = self._session.query(SampleContext.id, SampleContext.environment_id)
         subq = self._build_taxonomy_subquery()
         q = self._apply_filters(q, subq).order_by(SampleContext.id)
         return self._q_all_cached('matching_sample_ids_and_environment', q)
 
-    def matching_sample_headers(self, required_headers=None):
+
+    def matching_sample_headers(self, required_headers=None, sort_col=None, sort_order=None):
         query_headers = [SampleContext.id, SampleContext.environment_id]
+        joins = [] # Keep track of any foreign ontology classes which may be needed to be joined to.
 
         cache_name = ['matching_sample_headers']
         if required_headers:
             cache_name += required_headers
             for h in required_headers:
-                query_headers.append(getattr(SampleContext, h))
+                if h != '':
```

Comment:
I generally prefer keeping "skip" clauses like this out of the main flow; so I'd do:

```
if h != '': # (or maybe if not h)?
  continue
...
```

Reply:
Fixed.